libc/src/stdlib/memalignment.c

```diff
@@ -0,0 +1,19 @@
+#include "src/stdlib/memalignment.h"
+#include "src/__support/macros/config.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(size_t, memalignment, (const void *p)) {
+  if (p == NULL) {
+    return 0;
+  }
+
+  uintptr_t addr = (uintptr_t)p;
```

Comment:
```suggestion
  uintptr_t addr = static_cast<uintptr_t>(p);
```
Comment below seems overly verbose, this is a trivial function.